dotnet/samples/GettingStartedWithProcesses/Steps/CompleteNewCustomerFormStep.cs

```diff
@@ -0,0 +1,218 @@
+        """;
+
+    internal string _welcomeMessage = """
+        Hello there, I can help you out fill out the information needed to open a new account with us.
+        Please provide some personal information like first name and last name to get started.
+        """;
+
+    private readonly JsonSerializerOptions _jsonOptions = new()
+    {
+        DefaultIgnoreCondition = JsonIgnoreCondition.Never
+    };
+
+    public override ValueTask ActivateAsync(KernelProcessStepState<NewCustomerFormState> state)
+    {
+        _state = state.State ?? new();
+        _state.newCustomerForm ??= new();
+        return ValueTask.CompletedTask;
+    }
+
+    [KernelFunction(Functions.NewAccountWelcome)]
+    public async Task NewAccountWelcomeMessageAsync(KernelProcessStepContext context, Kernel _kernel)
+    {
+        _state?.conversation.Add(new ChatMessageContent { Role = AuthorRole.Assistant, Content = _welcomeMessage });
+        await context.EmitEventAsync(new() { Id = AccountOpeningEvents.NewCustomerFormWelcomeMessageComplete, Data = _welcomeMessage });
+    }
+
+    private Kernel CreateNewCustomerFormKernel(Kernel _baseKernel)
+    {
+        // Creating another kernel that only makes use private functions to fill up the new customer form
+        Kernel kernel = new(_baseKernel.Services);
+        kernel.ImportPluginFromFunctions("FillForm", [
+            KernelFunctionFactory.CreateFromMethod(OnUserProvidedFirstName, functionName: nameof(OnUserProvidedFirstName)),
+            KernelFunctionFactory.CreateFromMethod(OnUserProvidedLastName, functionName: nameof(OnUserProvidedLastName)),
+            KernelFunctionFactory.CreateFromMethod(OnUserProvidedDOBDetails, functionName: nameof(OnUserProvidedDOBDetails)),
+            KernelFunctionFactory.CreateFromMethod(OnUserProvidedStateOfResidence, functionName: nameof(OnUserProvidedStateOfResidence)),
+            KernelFunctionFactory.CreateFromMethod(OnUserProvidedPhoneNumber, functionName: nameof(OnUserProvidedPhoneNumber)),
+            KernelFunctionFactory.CreateFromMethod(OnUserProvidedUserId, functionName: nameof(OnUserProvidedUserId)),
+            KernelFunctionFactory.CreateFromMethod(OnUserProvidedEmailAddress, functionName: nameof(OnUserProvidedEmailAddress)),
+        ]);
+
+        return kernel;
+    }
+
+    [KernelFunction(Functions.NewAccountProcessUserInfo)]
+    public async Task CompleteNewCustomerFormAsync(KernelProcessStepContext context, string userMessage, Kernel _kernel)
+    {
+        // Keeping track of all user interactions
+        _state?.conversation.Add(new ChatMessageContent { Role = AuthorRole.User, Content = userMessage });
+
+        Kernel kernel = CreateNewCustomerFormKernel(_kernel);
+
+        OpenAIPromptExecutionSettings settings = new()
+        {
+            ToolCallBehavior = ToolCallBehavior.AutoInvokeKernelFunctions,
+            Temperature = 0.7,
+            ModelId = "gpt-4o",
```

Comment:
Would it be more flexible to retrieve model name from configuration?  (In Azure, the model deployment naming can vary as desired.)